From a55a3f45f55d87054ce055b159f8aaf5ecbdb439 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Mon, 6 Jul 2020 17:03:19 +0000
Subject: [PATCH 1022/1077] MANTIS 0057761 Darken Screen using the
 Navigation-Pad extend ComponentIF: prepareDevice, resetDevices,
 beginMediaRecord, switchEnableDevice extend PlatformIF: switchEnableDevice

switch: get switch result: on/off
prepareDevice/beginMediaRecord: do not switch off display in CLA
resetDevices: do not change display on activation transfer


git-svn-id: svn://localhost/SelfServiceCommon/trunk@1579 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 13a09183..346692ee 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -128,12 +128,12 @@ module PlatformManagement
         /** Prepares a device for a specific application context.
             @param contextName - The name of the application context
             @return RC_OK if the context switch was successfull, RC_ERROR if not. */
-        code prepareDevice(in string contextName);
+        code prepareDevice(in string contextName, in string appId, in boolean cla);
 
         /** Tries to bring the device in a proper state to be used by
             the next application.
             @return RC_OK is the reset was successfull, RC_ERROR if not. */
-        code resetDevices();
+        code resetDevices(in boolean softSwitch);
 
         /** Tells the device component, that an error(e.g. paper jam)has been recovered.
             @return RC_OK. */
@@ -153,7 +153,7 @@ module PlatformManagement
         /** prepare a device for using by CLA in media-roll-off mode.
             record events for playback after next prepareDevice
         */
-        code beginMediaRecord(in string contextName, in string token);
+        code beginMediaRecord(in string contextName, in string appId, in string token);
 
         /** same as resetDevices; except for mediaOffRoller components
             the mediaOffRoller components will playback the stored events
@@ -162,7 +162,7 @@ module PlatformManagement
         
         code enableDevice();
         code disableDevice();
-        code switchEnableDevice();
+        code switchEnableDevice(out boolean on);
     };
 
     enum DataStatusExtension
@@ -433,7 +433,7 @@ module PlatformManagement
         
         code enableDisplay();
         code disableDisplay();
-        code switchEnableDisplay();
+        code switchEnableDisplay(out boolean on);
     };
 };
 
-- 
2.41.0.windows.1

